### PR TITLE
feat: allow third click to disable sorting (fixes #132)

### LINKED ISSUE #132: Allow a third click to disable sorting
Right now, once you click to sort issues by some criteria there is no way to unsort them. Usually apps have a three clicks cycle where first click sorts top to bottom, second: bottom to top and a third disables the sorting. Not having this is not harmful but it's a worthy detail. Also we could eventually implement a standard sorting (priority based or paid placement by our partners) and we would want to let the user see it more than once.

https://github.com/user-attachments/assets/bda2629d-41c2-418e-a909-1cdb734409dc

### PR DESCRIPTION
**Pull request by Atharva-Kanherkar**
Resolves #132 

Please see the video below : 
 
[Ubuiquity_DAOissue1.webm](https://github.com/user-attachments/assets/4b8207f5-340d-4561-b040-622b8820bd85)

### CODE REVIEW

**Comment by zugdev** [comment]
Hey, the none option is working but you have to handle when user clicks on another sorting handle as well. Sorting is correct in that case but it should not display the asc/desc indicator

image

**Comment by zugdev** [comment]
These lines previously did that by clearing the data-sorted attribute for all other options 

https://github.com/ubiquity/work.ubq.fi/blob/cb5e4ad8b3e71768074968e9f64a1f2c8812dd80/src/home/sorting/sorting-manager.ts#L139-L143

**Comment by zugdev** [comment]
Also the current sorting CSS looks weird, @0x4007. The gray background toggle seems binary, we should either have it be gray when it's not "none" or not have this gray background ever.

image

image

**Comment by zugdev** [comment]
The rest of it looks good to me @Atharva-Kanherkar, just please make sure you clear all other options states on click.

**Comment by Atharva-Kanherkar** [comment]
> These lines previously did that by clearing the data-sorted attribute for all other options 
> 
> https://github.com/ubiquity/work.ubq.fi/blob/cb5e4ad8b3e71768074968e9f64a1f2c8812dd80/src/home/sorting/sorting-manager.ts#L139-L143

Oh okay! I might have missed that, thank you for mentioning it. I'll review and commit some changes!

**Comment by Atharva-Kanherkar** [comment]
> Also the current sorting CSS looks weird, @0x4007. The gray background toggle seems binary, we should either have it be gray when it's not "none" or not have this gray background ever.
> 
> image
> 
> image
> 
> 
So you want me to fix this in this PR? or this is just intended as a discussion?

**Comment by Atharva-Kanherkar** [comment]
Hey @zugdev I have added the changes!:) Now you can see it works as expected <3

**Comment by zugdev** [comment]
Looks good to me, just wait for code owner to review

**Comment by Atharva-Kanherkar** [comment]
@0x4007 Even after optimizing the code, The tests fail, is there a possibility that we can increase the timeouts in cypress?

**Comment by zugdev** [comment]
> @0x4007 Even after optimizing the code, The tests fail, is there a possibility that we can increase the timeouts in cypress?

Cypress is broken and up to be fixed at #109, if you want to fix them... But don't worry much for this PR I guess.

File: src/home/sorting/sorting-manager.ts
Comment (0x4007) [inline src/home/sorting/sorting-manager.ts]: Why do you debounce the input? I don't think this is normally a problem?

File: yarn.lock
Comment (0x4007) [inline yarn.lock:1]: Maybe you should regenerate your lock file. I don't see why it should have changed in this pull.

File: src/home/sorting/sorting-manager.ts
Comment (0x4007) [inline src/home/sorting/sorting-manager.ts]: I think we should remove the debounce logic.

**Comment by 0x4007** [comment]
Feels nice though I just tested it. I think when you address my comments I can merge!

File: src/home/sorting/sorting-manager.ts
Comment (zugdev) [reply src/home/sorting/sorting-manager.ts]: he was trying to make cypress work haha

Comment (Atharva-Kanherkar) [reply src/home/sorting/sorting-manager.ts]: I was trying to optimize some code because I thought the tests were failing because of me. I will remove the logic ASAP

File: yarn.lock
Comment (Atharva-Kanherkar) [reply yarn.lock:1]: Okay, I still dont understand why it happened. I will change it

File: src/home/sorting/sorting-manager.ts
Comment (Atharva-Kanherkar) [reply src/home/sorting/sorting-manager.ts]: yes, ill remove it

File: yarn.lock
Comment (Atharva-Kanherkar) [reply yarn.lock:1]: regenerating yarn.lock, but still appears in the pull, what can be the reason?

File: src/home/sorting/sorting-manager.ts
Comment (0x4007) [inline src/home/sorting/sorting-manager.ts:158]: Are you going to delete?

Comment (zugdev) [reply src/home/sorting/sorting-manager.ts:158]: This was old code, can remove that safely

**Comment by Atharva-Kanherkar** [comment]
@zugdev @0x4007 Any suggestions on how can I remove the yarn.lock from the pull ? Is this mergeable?

**Comment by zugdev** [comment]
> @zugdev @0x4007 Any suggestions on how can I remove the yarn.lock from the pull ? Is this mergeable?

You don't have to remove yarn.lock from the pull, just make sure you ran yarn and commited the appropriate yarn.lock file. You haven't installed any new packages and should be good to go

**Comment by Atharva-Kanherkar** [comment]
> > @zugdev @0x4007 Any suggestions on how can I remove the yarn.lock from the pull ? Is this mergeable?
> 
> You don't have to remove yarn.lock from the pull, just make sure you ran yarn and commited the appropriate yarn.lock file. You haven't installed any new packages and should be good to go

So acc to u it seems mergeable?

File: yarn.lock
Comment (zugdev) [reply yarn.lock:1]: different OSs generate different lock files

Comment (zugdev) [reply yarn.lock:1]: This might be a problem with yarn on itself https://github.com/yarnpkg/yarn/issues/4379

**Review by 0x4007 — APPROVED** [review]

**Pull request merged**
